### PR TITLE
use "files" prop of package.json; remove .npmignore

### PR DESCRIPTION
**Pull request by boneskull**
This makes the list of files to be published a *whitelist* instead of a *blacklist* (`.npmignore`).  This is typically a safer option.  Effectively, there will be no change to the list of files which are published.

`README.md`, `LICENSE` and `package.json` are automatically published by npm.

* * *

This is a refactor (if that).  Anecdotally, I've found the `files` prop of `package.json` to be safer than just relying on `.npmignore`.  This may very well be a "best practice," but I have no references.

In a nutshell, the implications are that of changing any hypothetical permission from "deny" to "allow". 

I didn't feel like anyone would feel strongly about this change (or at least any more strongly than I do), so I did not raise it on the mailing list.

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -1,7 +0,0 @@
-.settings
-.jshintignore
-.jshintrc
```
Comment (dceejay) [inline .npmignore]: We still want some style hints

Comment (knolleary) [reply .npmignore]: And this pr isn't removing them... its changing how npm knows what files to publish when we do a release.

Comment (dceejay) [reply .npmignore]: ok - so still in for a git clone - but not for an npm pack.

Comment (knolleary) [reply .npmignore]: npm pack obeys `.npmignore` - so these files are already not included today. This PR doesn't change that fact.

Comment (dceejay) [reply .npmignore]: ok - happy to accept.
